src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java

```diff
@@ -485,65 +579,75 @@ public LockedContext<T> lockAndGetExecutor(final String id) {
         int shardId = shardCalculator.shardId(id);
         LookupDaoPriv dao = daos.get(shardId);
         return new LockedContext<>(shardId, dao.sessionFactory, () -> dao.getLockedForWrite(id),
-                entityClass, shardInfoProvider, observer);
+                                   entityClass, shardInfoProvider, observer);
+    }
+
+    public ReadOnlyContext<T> readOnlyExecutor(String id) {
+        return readOnlyExecutor(id, x -> x);
     }
 
     /**
      * Creates and returns a read-only context for executing read operations on an entity with the specified ID.
-     *
+
      * <p>This method calculates the shard ID based on the provided entity ID, retrieves the LookupDaoPriv
      * for the corresponding shard, and creates a read-only context for executing read operations on the entity.
      * It does not perform entity population during read operations.
-     *
+
      * @param id The ID of the entity for which the read-only context is created.
      * @return A new ReadOnlyContext for executing read operations on the specified entity.
      */
-    public ReadOnlyContext<T> readOnlyExecutor(final String id) {
+    public ReadOnlyContext<T> readOnlyExecutor(String id, UnaryOperator<Criteria> criteriaUpdater) {
```

Comment:
either add criteriaUpdater in documentation or move public ReadOnlyContext<T> readOnlyExecutor(String id) below documentation

Reply:
done